pkg/infra/main.go

```diff
@@ -26,10 +29,32 @@ const monsterHealth = 100
 const monsterAttack = 240
 
 func main() {
+	// define flags
+	configFile := flag.String("c", "config.json", "configuration file")
+	useJSONFormatter := flag.Bool("j", false, "whether to use JSONFormatter for logging")
+	flag.Parse()
+
+	initLogger(*useJSONFormatter)
+
+	log.Debug("read configuration from file: ", *configFile)
+
 	agentMap, stateChannels, decisionChannels, globalState := initialise()
 	gameLoop(globalState, agentMap, decisionChannels, stateChannels)
 }
 
+func initLogger(useJSONFormatter bool) {
```

Comment:
Should we move this function to be outside of the main.go file to keep it as clean as possible? Maybe we can put it in a infra/logging folder or something instead?

Reply:
sure, will do it in the next commit